### PR TITLE
Added basic handling for invalid plan amount

### PR DESCRIPTION
**Pull request by allouis**
refs https://github.com/TryGhost/Ghost/issues/11765

This adds error handling to the stripe plan prices to ensure they are set for at least 1 USD (or other currency) or more.

### CODE REVIEW

**Comment by allouis** [comment]
Hey @peterzimon i've updated the subscription inputs to display the error, but had some issues getting the margins to work - would you be able to take a look?


image

I had to get rid of the margins to make the inputs sit in the same flex box row :grimacing: Not sure if that's correct or not

**Comment by allouis** [comment]
Whoooops - forgot Zimo is off the rest of the week, @kevinansfield Is this something you might be able to help with?

**Comment by allouis** [comment]
So I think there's a few questions here that need answering:

1. Is the error text correct/okay?
2. How do we get the inputs to have the red border around the whole thing?
3. How can we get the error message to sit below both inputs?

@kevinansfield You might be able to help with 2. and 3. - if you have some time?

**Comment by kevinansfield** [comment]
> How do we get the inputs to have the red border around the whole thing?

A new style needs to be added:

```
.error .gh-input-append {
    border-color: var(--red);
}
```

File: app/components/gh-members-lab-setting.js [hunk 1]
```diff
@@ -93,11 +96,11 @@ export default Component.extend({
 
         return {
             monthly: {
-                amount: parseInt(monthly.amount) / 100 || 0,
+                amount: parseInt(monthly.amount) / 100 || 1,
                 currency: monthly.currency
             },
             yearly: {
-                amount: parseInt(yearly.amount) / 100 || 0,
+                amount: parseInt(yearly.amount) / 100 || 1,
```
Comment (matthanley) [inline app/components/gh-members-lab-setting.js:103]: Should default values for monthly/yearly match [default settings](https://github.com/TryGhost/Ghost/blob/master/core/server/data/schema/default-settings.json#L249) of 5/50?

**Comment by allouis** [comment]
I've updated this to make less changes to the HTML, but i've reintroduced the spacing bug when theres an error:

With no error:
image
 
With error:
image

File: app/components/gh-members-lab-setting.hbs [hunk 2]
```diff
@@ -171,32 +171,35 @@
...
-                            @input={{action "setStripePlan" "year"}}
+                            @input={{action (mut this._scratchStripeYearlyAmount) value="target.value"}}
+                            @focus-out={{action "validateStripePlans"}}
                         />
                         <span class="gh-input-append"><span class="ttu">{{this.stripePlans.yearly.currency}}</span>/year</span>
                     </div>
                     </GhFormGroup>
                 </div>
+                <GhErrorMessage @errors={{settings.errors}} @property="stripePlans" class="w-100 red"/>
             </div>
```
Comment (matthanley) [inline app/components/gh-members-lab-setting.hbs:203]: This looks like it fixes it to me, but I'm not a frontend dev 😬 

```suggestion
            </div>
            <div class="w-100 w-50-l flex flex-column flex-row-ns">
                <GhErrorMessage @errors={{settings.errors}} @property="stripePlans" class="w-100 red"/>
            </div>
```

image

also
* nitpick: 1 is valid, but the error says "greater than 1"
* might be cool to add the currency in the error: "Please enter a value greater than $1.00"

pinging @JohnONolan for feedback

**Comment by allouis** [comment]
@matthanley Added your changes and improved the error message!

**Pull request merged**